### PR TITLE
Remove InferReturnTypeComponents test pattern from hlo_test_infer.

### LINKED ISSUE #938: Consider removing hlo_test_infer.get_return_type_components.
Remove all tests that depend on hlo_test_infer.get_return_type_components because they can be now covered using hlo_test_infer.get_return_types instead.

### PR DESCRIPTION
**Pull request by subhankarshah**
* Move all usages of get_return_type_componets to get_return_types.
  * Move `CHECK` directives in tests before the corresponding MLIR code.

closes #938

### CODE REVIEW

File: stablehlo/tests/infer_stablehlo.mlir [hunk 1]
```diff
@@ -19,9 +19,9 @@ func.func @select(%pred : tensor<i1>, %a : tensor<?x2x3xf32>, %b : tensor<1x?x3x
     -> tensor<1x2x3xindex> {
   %0 = "stablehlo.select"(%pred, %a, %b)
       : (tensor<i1>, tensor<?x2x3xf32>, tensor<1x?x3xf32>) -> tensor<*xf32>
-  %1 = "hlo_test_infer.get_return_type_components"(%0)
+  %1 = "hlo_test_infer.get_return_types"(%0)
       : (tensor<*xf32>) -> tensor<1x2x3xindex>
-  // CHECK: %1 = "hlo_test_infer.return_type_components"(%0) {dims0 = "[1, 2, 3]", element_type0 = f32} : (tensor<*xf32>) -> tensor<1x2x3xindex>
+  // CHECK: {types0 = tensor<1x2x3xf32>}
```
Comment (ghpvnist) [inline stablehlo/tests/infer_stablehlo.mlir:24]: The CHECK directive is inconsistently above and below the %1. Can you stick to one version? Here and below.

**Review by ghpvnist — APPROVED** [review]
Minor comment.

**Review by ghpvnist — APPROVED** [review]

File: stablehlo/tests/infer_stablehlo.mlir [hunk 2]
```diff
@@ -543,8 +523,8 @@ func.func @select_and_scatter(
     window_strides = dense<[1, 2, 2, 1]> : tensor<4xi64>
   } : (tensor<10x24x24x64xf32>, tensor<10x12x12x64xf32>, tensor<f32>) ->
         tensor<10x24x24x64xf32>
+  // CHECK: %2 = "hlo_test_infer.return_types"(%1) {types0 = tensor<10x24x24x64xf32>} : (tensor<10x24x24x64xf32>) -> tensor<10x24x24x64xindex>
```
Comment (burmako) [inline stablehlo/tests/infer_stablehlo.mlir:526]: In the majority of this PR, we're using a simplified version of this CHECK, where only `typesN` are called out. Let's migrate everything to consistently do that?

Comment (burmako) [inline stablehlo/tests/infer_stablehlo.mlir:526]: Also can we simplify `{types0 = tensor<10x24x24x64xf32>}` to just `types0 = tensor<10x24x24x64xf32>`? That would make things easier to read and write.

**Review by burmako — CHANGES_REQUESTED** [review]

**Review by burmako — APPROVED** [review]

**Pull request merged**